src/opensles/AudioStreamOpenSLES.h

```diff
@@ -64,8 +61,23 @@ class AudioStreamOpenSLES : public AudioStreamBuffered {
 
     int32_t getFramesPerBurst() override;
 
+    static SLuint32 getDefaultByteOrder();
+
+    virtual int chanCountToChanMask(int chanCount) = 0;
```

Comment:
Why this method needs to be declared as part of the interface? It doesn't seem to be used in any common code, only in particular implementations. Also, it can be made static.

Reply:
I made it protected. It was already static. It is used by input and output streams to open a stream.

Reply:
Sorry if I didn't explain my point clear enough. What I was trying to say is that virtual methods in the base class usually exist because there is either an external client or common code in the base class that uses them. For chanCountToChanMask, the usages only seem to be in the derived classes (in and out streams). So why not just define these methods in the derived classes?

Putting chanCountToChanMask doesn't help with code reuse, it only pollutes the interface definition.